pkg/collector/corechecks/cluster/kubernetes_apiserver.go

```diff
@@ -66,8 +66,13 @@ func (c *KubeASConfig) parse(data []byte) error {
 
 // Configure parses the check configuration and init the check.
 func (k *KubeASCheck) Configure(config, initConfig integration.Data) error {
+	err := k.CommonConfigure(config)
```

Comment:
I'm afraid this would be forgotten, for example in the CRI Core Check, if you're not aware of it

Could we find a clever way to enforce it?

Reply:
I agree, there's a risk, that's why I doubled down on the documentation.

Golang's limited encapsulation paradigm makes it pretty hard to enforce with the current design: we cannot have `CheckBase.Configure()` call a check's `customConfigure()`, as if it was an abstract method

Any idea on how we could make it work?

Reply:
One of the only way I can think of would be to add a new method to the interface (like `Setup` or `CommonConfigure` that will likely not get overriden and implemented in the check base class

Not sure it's worth it though